File: UI/wwwroot/Model/Familias.js
//@ts-check
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
//@ts-ignore
import { ModelProperty } from "../WDevCore/WModules/CommonModel.js";
import { Estudiantes_responsables_familia }  from './Estudiantes_responsables_familia.js'
//import { Security_Users }  from './Security_Users.js'

class Familias extends EntityClass {
   /** @param {Partial<Familias>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Idtfamilia;
   /**@type {String}*/ Descripcion;
   /**@type {Boolean}*/ Estado;
   /**@type {Number}*/ Saldo;
   /**@type {Boolean}*/ Actualizado;
   /**@type {Boolean}*/ Aceptacion;
   /**@type {Number}*/ Periodo_aceptacion;
   /**@type {Date}*/ Fecha_actualizacion;
   /**@type {String}*/ Fecha_ultima_notificacion;
   //**@type {Security_Users} ManyToOne*/ Security_Users;
   /**@type {Array<Estudiantes_responsables_familia>} OneToMany*/ Estudiantes_responsables_familia;
}
export { Familias }

File: CAPA_NEGOCIO/Gestion_Responsables/Model/Familias.cs
using APPCORE;
using APPCORE.Security;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Familias : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id { get; set; }
       public string? Idtfamilia { get; set; }
       public string? Descripcion { get; set; }
       public bool? Estado { get; set; }
       public Double? Saldo { get; set; }
       public bool? Actualizado { get; set; }
       public bool? Aceptacion { get; set; }
       public int? Periodo_aceptacion { get; set; }
       public DateTime? Fecha_actualizacion { get; set; }
       public DateTime? Fecha_ultima_notificacion { get; set; }

       [OneToMany(TableName = "Estudiantes_responsables_familia", KeyColumn = "Id", ForeignKeyColumn = "Familia_id")]
       public List<Estudiantes_responsables_familia>? Estudiantes_responsables_familia { get; set; }
   }
}
